src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionAnswerResource.java

```diff
@@ -194,29 +172,24 @@ else if (studentQuestionAnswer.getQuestion().getExercise() != null) {
             return ResponseEntity.badRequest().build();
         }
         if (!course.getId().equals(courseId)) {
-            return forbidden();
-        }
-        if (mayUpdateOrDeleteStudentQuestionAnswer(studentQuestionAnswer, user)) {
-            log.info("StudentQuestionAnswer deleted by {}. Answer: {} for {}", user.getLogin(), studentQuestionAnswer.getAnswerText(), entity);
-            studentQuestionAnswerRepository.deleteById(id);
-            return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, id.toString())).build();
-        }
-        else {
-            return forbidden();
+            return badRequest("courseId", "400", "PathVariable courseId doesnt match courseId of the StudentQuestionAnswer that should be deleted");
         }
+        mayUpdateOrDeleteStudentQuestionAnswerElseThrow(studentQuestionAnswer, user);
+        log.info("StudentQuestionAnswer deleted by " + user.getLogin() + ". Answer: " + studentQuestionAnswer.getAnswerText() + " for " + entity, user.getLogin());
```

Comment:
the second argument is not needed anymore

```suggestion
        log.info("StudentQuestionAnswer deleted by " + user.getLogin() + ". Answer: " + studentQuestionAnswer.getAnswerText() + " for " + entity);
```